src/ui/cards-post/card-post.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { AverageCard } from '../cards/average-card/average-card';
+import { SmallCard } from '../cards/small-card/small-card';
+import { LikeDislike } from '../../features/like-dislike/like-dislike';
+import { mokieApi } from '../../mokie.api';
+import { Link } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import {
+  setIsModalOpen,
+  setPreviewImg,
+} from '../../features/preview-pop-up/preview-pop-up.slice';
+import closeBtn from '../../ui/menu/img/close_menu.svg';
+
+import { PreviewPopUp } from '../../features/preview-pop-up/preview-pop-up';
+
+export const CardPost: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const isModalOpen = useAppSelector((state) => state.previewPopUp.isModalOpen);
+  const previewImg = useAppSelector((state) => state.previewPopUp.image);
+
+  return (
+    <CardPostW>
+      <PostWrapper>
+        <PostLeft>
+          <LeftBottomWrapper>
+            <LeftTop>
+              {[1, 2, 3].map((el) => (
+                <StyledLink to={`/posts/${el + 1}`}>
+                  <AverageCard
+                    id={mokieApi[el].id}
+                    key={mokieApi[el].id}
+                    image={<img src={mokieApi[el].image} alt="astronaut"></img>}
+                    text={<div>{mokieApi[el].text}</div>}
+                    date={<div>{mokieApi[el].date}</div>}
+                    title={<div>{mokieApi[el].title}</div>}
+                    LikeDislike={LikeDislike}
+                    setActive={() => dispatch(setIsModalOpen(true))}
+                    setImg={() => dispatch(setPreviewImg(mokieApi[el].image))}
+                  ></AverageCard>
+                  <PreviewPopUp
+                    active={isModalOpen}
+                    setActive={() => dispatch(setIsModalOpen)}
```

Comment:
UI can't dispatch redux actions; move the card to the features or "pass" redux logic to prop via onClick prop